### PR TITLE
SOLR-16420: {!mlt_content}

### PR DESCRIPTION
**Pull request by mkhludnev**
https://issues.apache.org/jira/browse/SOLR-16420

~TODO: add cloud test and equality test, if necessary and indent the code~  **Done**.

One question: `mindf` default for cloud test is 0 in code now that contradicts to the ref guide - 5. This PR sets default mindf to 5. It is a good idea to fix it by the PR?

### CODE REVIEW

**Comment by mkhludnev** [comment]
I think it's ready. The only question is fixing mindf default for cloud `{!mlt}`

File: solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    expectedIds = new int[] {31, 18, 23, 13, 14, 20, 22, 32, 19, 21};
+    actualIds = new int[solrDocuments.size()];
+    i = 0;
+    for (SolrDocument solrDocument : solrDocuments) {
+      actualIds[i++] = Integer.parseInt(String.valueOf(solrDocument.getFieldValue("id")));
+    }
+
+    Arrays.sort(actualIds);
+    Arrays.sort(expectedIds);
+    System.out.println("DEBUG ACTUAL IDS 1: " + Arrays.toString(actualIds));
```
Comment (risdenk) [inline solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java:123]: leftover debugging?

File: solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    expectedIds = new int[] {29, 31, 32, 18, 23, 13, 14, 20, 22, 19};
+    actualIds = new int[solrDocuments.size()];
+    i = 0;
+    for (SolrDocument solrDocument : solrDocuments) {
+      actualIds[i++] = Integer.parseInt(String.valueOf(solrDocument.getFieldValue("id")));
+    }
+
+    Arrays.sort(actualIds);
+    Arrays.sort(expectedIds);
+    System.out.println("DEBUG ACTUAL IDS 2: " + Arrays.toString(actualIds));
```
Comment (risdenk) [inline solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java:147]: left over debugging?

File: solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    assertArrayEquals(
+        Arrays.toString(expectedIds) + " " + Arrays.toString(actualIds), expectedIds, actualIds);
```
Comment (risdenk) [inline solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java:149]: not sure that `Arrays.toString(expectedIds) + " " + Arrays.toString(actualIds),` is needed? I think assertArrayEquals prints the differences.

**Review by risdenk — APPROVED** [review]
looks good just minor comment about leftover system.out debugging.

**Comment by mkhludnev** [comment]
OK. Thanks, @risdenk I'll sweep those leftovers.

File: solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java (hunk 2 above)
Comment (mkhludnev) [reply solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java:147]: not mine

**Pull request merged**